AIFG-3D-Connect-4/ConsoleInput.cpp

```diff
@@ -0,0 +1,34 @@
+#include "ConsoleInput.h"
+
+ConsoleInput::ConsoleInput(GameBoard& t_board) : m_board(t_board)
+{}
+
+Coordinate ConsoleInput::getCoordinate()
+{
+	Coordinate pos{ 0, 0, 0 }; // set the coordinates to a default value.
+	bool validInput = false;
+	do
+	{
+		std::cout << "Enter your move: ";
+		std::cin >> pos.x >> pos.y >> pos.z;
+		if (std::cin.fail() || validateIfInputWithinGameBoard(pos)) // checks if the input is valid.
+		{
+			pos = { 0, 0, 0 };
+			std::cin.clear();
+			std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+			continue;
+		}
+		else
+		{
+			validInput = true; // if the input has gone through vaildation, then the input is vaild.
+			continue;
+		}
+	} while (PieceType::None != m_board.getPiece(pos) || !validInput); // Loop, until you find pos that is valid.
```

Comment:
I wouldn't check if the board position is available here, let the rules handler do that. It can keep prompting for input if the inputted location is already filled.
The input doesn't need a reference to the board.